experimental/Geometry/Misc.jl

```diff
@@ -0,0 +1,202 @@
+import AbstractAlgebra.Ring
+import AbstractAlgebra.RingElement
+import Oscar: radical_membership
+
+
+module Misc
+using Oscar
+
+export add_variables, divides_power
+
+####################################################################
+#  
+#  Miscellaneous routines used in the above 
+#
+####################################################################
+
+
+function add_variables( R::T, new_vars::Vector{String} ) where{ T<:MPolyRing } 
+  k = base_ring(R)
+  old_vars = String.( symbols(R) )
+  n = length( old_vars )
+  vars = vcat( old_vars, new_vars )
+  S, v = PolynomialRing( k, vars )
+  phi = AlgebraHomomorphism( R, S, gens(S)[1:n] )
+  y = v[n+1:length(v)]
+  return S, phi, y
+end
+
+function add_variables( P::T, new_vars::Vector{String} ) where{ T<:MPolyQuo } 
+  R = base_ring(P)
+  k = base_ring(R)
+  S, phi, y = add_variables( R, new_vars )
+  I = ideal( S, [ phi(g) for g in gens(modulus(P)) ] )
+  Q, projection = quo( S, I )
+  phi_quot = AlgebraHomomorphism( P, Q, [projection(g) for g in phi.image] )
+  return Q, phi_quot, [ projection(g) for g in y]
+end
+
+####################################################################
+#
+# Test whether some power of the polynomial f is contained in the 
+# principal ideal generated by g
+#
+function Oscar.radical_membership( f::U, g::U ) where{ U<:MPolyElem }
+  R = parent(f)
+  if R != parent(g) 
+    error( "Polynomials do not belong to the same ring" )
+  end
```

Comment:
```suggestion
  R == parent(g) || error("Polynomials do not belong to the same ring")
```
... and similar in many other places